clients/src/main/java/org/apache/kafka/common/record/Records.java

```diff
@@ -90,4 +91,18 @@ public interface Records extends TransferableRecords {
      * @return The record iterator
      */
     Iterable<Record> records();
+
+    /**
+     * Return a slice of records from this instance, which is a view into this set starting from the given position
+     * and with the given size limit.
+     *
+     * If the size is beyond the end of the records, the end will be based on the size of the records at the time of the read.
+     *
+     * If this records set is already sliced, the position will be taken relative to that slicing.
+     *
+     * @param position The start position to begin the read from
```

Comment:
Could we add that position is expected to be aligned at the batch boundary? Otherwise, the returned Records can't be iterated.

Reply:
Done.